### PR TITLE
minikube 0.26.1 (new formula)

### PR DESCRIPTION
**Pull request by chases**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Continuation of #12189

### CODE REVIEW

File: Formula/minikube.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Minikube < Formula
+  desc "Tool that makes it easy to run Kubernetes locally"
+  homepage "https://kubernetes.io/docs/getting-started-guides/minikube/"
+  url "https://github.com/kubernetes/minikube.git",
```
Comment (fxcoudert) [inline Formula/minikube.rb]: Can we build from source archive? `https://github.com/kubernetes/minikube/archive/v0.26.1.tar.gz`

File: Formula/minikube.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    :tag => "v0.26.1",
+    :revision => "6ded2b647d1b1f62100c630bcfcc1363c631ce2d"
+
+  # Docker required for non-Linux builds
```
Comment (fxcoudert) [inline Formula/minikube.rb]: Homebrew is for macOS. Remove `unless MacOS.version == Version::NULL` and comment

File: Formula/minikube.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "docker" => :build unless MacOS.version == Version::NULL
+
```
Comment (fxcoudert) [inline Formula/minikube.rb]: No blank line

File: Formula/minikube.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    repo_path.install Dir["*"]
+
+    cd repo_path do
+      system "make"
+      bin.install "out/minikube" => "minikube"
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/minikube version 2>&1")
```
Comment (fxcoudert) [inline Formula/minikube.rb]: We need a test that exercises the some of the functionality of the app.

File: Formula/minikube.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    repo_path = buildpath/"src/k8s.io/minikube"
+    repo_path.install Dir["*"]
+
+    cd repo_path do
+      system "make"
+      bin.install "out/minikube" => "minikube"
```
Comment (fxcoudert) [inline Formula/minikube.rb]: No doc, no man page, no other file to install?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/minikube.rb (hunk 2 above)
Comment (chases) [reply Formula/minikube.rb]: This formula is meant to work on Linuxbrew as well, how is that normally handled?

File: Formula/minikube.rb (hunk 1 above)
Comment (chases) [reply Formula/minikube.rb]: The only problem with that is the make script depends on the build directory having a valid git repository.

File: Formula/minikube.rb (hunk 4 above)
Comment (chases) [reply Formula/minikube.rb]: If you don't mind, would you point me to an example you'd like me to follow? Lots of the formulas I looked at do more or less exactly this.

File: Formula/minikube.rb (hunk 2 above)
Comment (fxcoudert) [reply Formula/minikube.rb]: Linuxbrew will handle the changes for Linux

File: Formula/minikube.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/minikube.rb]: OK then

File: Formula/minikube.rb (hunk 4 above)
Comment (fxcoudert) [reply Formula/minikube.rb]: Look at `kube-aws` or `kubeless`

File: Formula/minikube.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    repo_path = buildpath/"src/k8s.io/minikube"
+    repo_path.install Dir["*"]
+
+    cd repo_path do
+      system "make"
```
Comment (commitay) [inline Formula/minikube.rb]: This is running `go get`

**Comment by commitay** [comment]
Upstream has an [open issue](https://github.com/kubernetes/minikube/issues/2141) for creating a Homebrew formula.

> we can create a real brew formula for minikube now that we no longer include (and therefore don't cross build) localkube with go-bindata

Currently localkube is still included but it has a PR open for it to be removed. https://github.com/kubernetes/minikube/pull/2736

**Comment by fxcoudert** [comment]
Hi @chases, thanks for the pull request, we are going to close it for now. We would be happy to consider a revised formula that does not rely on `go get` (which is not allowed inside Homebrew core) and has a functional test.

**Pull request closed without merging**